src/foam/demos/net/nap/web/model/Messageboard.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.demos.net.nap.web.model',
+  name: 'Messageboard',
+
+  tableColumns: [
+    'mark', 'title', 'createdDate', 'creator', 'attachment', 'hits'
+  ],
+
+  imports: [
+    'blobService',
+    'messageboardDAO',
+    'user'
+  ],
+
+  javaImports: [ 'java.util.Date' ],
+
+  requires: [
+    'foam.blob.BlobBlob',
+    'foam.nanos.fs.File',
+    'foam.u2.dialog.NotificationMessage'
+  ],
+
+  properties: [
+    {
+      class: 'Long',
+      name: 'id'
+    },
+    {
+      class: 'Boolean',
+      name: 'mark',
+      tableCellFormatter: function(value, obj, property) {
+        this
+          .start()
+            .call(function() {
+              if ( value ) { this.tag({ class: 'foam.u2.tag.Image', data: 'images/star.svg' }).style({'width': '25'}); }
+            })
+          .end();
+      }
+    },
+    {
+      class: 'String',
+      name: 'title',
+      displayWidth: 100,
+      width: 200
+    },
+    {
+      class: 'String',
+      name: 'creator',
+      visibility: foam.u2.Visibility.RO,
+      factory: function() { return this.user.firstName; }
+    },
+    {
+      class: 'Long',
+      name: 'creatorId',
+      visibility: foam.u2.Visibility.RO,
+      factory: function() { return this.user.id; },
```

Comment:
This will need also need to be set or verified on the server to prevent people from making fake messages.